analyzer/codechecker_analyzer/analyzers/cppcheck/analyzer.py

```diff
@@ -83,6 +83,29 @@ def get_analyzer_mentioned_files(self, output):
         """
         pass
 
+    def parse_analyzer_config(self):
+        "Parses a white listed compiler flags"
+        params = []
+        interesting_option = re.compile("^-[I|U|D].*$")
+        for i, analyzer_option in enumerate(self.buildaction.analyzer_options):
+            if interesting_option.match(analyzer_option):
+                params.extend([analyzer_option])
+                # The above extend() won't properly insert the analyzer_option
+                # in case of the following format -I <path/to/include>.
+                # The below check will add the next item in the
+                # analyzer_options list if the parameter is specified with a
+                # space, as that should be actual path to the include.
+                if interesting_option.match(analyzer_option).span() == (0, 2):
+                    params.extend(
+                        [self.buildaction.analyzer_options[i+1]]
+                    )
+            if analyzer_option.startswith("-std"):
+                standard = analyzer_option.split("=")[-1] \
+                    .lower().replace("gnu", "c")
+                params.extend(["--std=" + standard])
+        LOG.debug(params)
```

Comment:
I think we should add some label to it so we know what is printed here. I also think that we should print debug logs only if we can gain information from it when some debugging is needed with the users. I think we should eliminate debug logs that are only useful during development. The assembled build command can be seen even now in debug logs.

Reply:
True, this is a leftover development debug print.